api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -158,6 +164,39 @@ public String generateHtml(FormEntryContext context) {
             }
         }
     }
+    
+    /**
+     * This method recieves a comma seperated list of diagnosis sets which are verified and then
+     * returns as comma seperated list of their uuids. diagnosisSetIds may be concept's 1). uuid
+     * 2). mapping, 3). id
+     *
+     * @param diagnosisSetIds
+     * @return uuids
+     */
+    private String validateAndFormat(String diagnosisSetIds) {
+        if ("".equals(diagnosisSetIds)) {
+            return diagnosisSetIds;
+        }
+        List<Concept> concepts = new ArrayList<Concept>();
+        for (StringTokenizer st = new StringTokenizer(diagnosisSetIds, ","); st.hasMoreTokens();) {
+            String id = st.nextToken().trim();
+            Concept concept = HtmlFormEntryUtil.getConcept(id);
+            if (concept == null) {
+                throw new IllegalArgumentException("Cannot find diagnosis set for value " + id
+                        + " in diagnosisSets attribute value. Parameters: " + diagnosisSetIds);
+            }
+            concepts.add(concept);
+        }
+        StringBuilder sb = new StringBuilder("");
+        if (CollectionUtils.isNotEmpty(concepts)) {
+            for (Concept con : concepts) {
+                sb.append(con.getUuid() + ",");
+            }
+        }
+        String uuids = sb.toString();
+
+        return uuids.substring(0, uuids.lastIndexOf(","));
```

Comment:
To make this a little safer, have a look at this: https://stackoverflow.com/a/15900408/321797